test/cli/name-pubsub.js

```diff
@@ -0,0 +1,253 @@
+              expect(res).to.exist()
+              expect(res).to.have.string('canceled') // canceled now
+
+              ipfsA('pubsub ls').then((res) => {
+                expect(res).to.exist()
+                expect(res).to.not.have.string('/ipns/') // ipns subscribtion not available
+
+                ipfsA('name pubsub subs').then((res) => {
+                  expect(res).to.exist()
+                  expect(res).to.not.have.string(`/ipns/${nodeBId.id}`) // ipns subscribtion not available
+
+                  done()
+                })
+              })
+            })
+          })
+        })
+      })
+    })
+
+    describe('pubsub records', () => {
+      let cidAdded
+
+      before(function (done) {
+        this.timeout(50 * 1000)
+        ipfsA('add src/init-files/init-docs/readme').then((out) => {
+          cidAdded = out.split(' ')[1]
+          done()
+        })
+      })
+
+      it('should publish the received record to the subscriber', function (done) {
+        this.timeout(80 * 1000)
+
+        ipfsB(`name resolve ${nodeBId.id}`).then((res) => {
+          expect(res).to.exist()
+          expect(res).to.satisfy(checkAll([emptyDirCid])) // Empty dir received (subscribed)
+
+          ipfsA(`name resolve ${nodeBId.id}`).catch((err) => {
+            expect(err).to.exist() // Not available (subscribed now)
+
+            ipfsB(`name publish ${cidAdded}`).then((res) => {
+              // published to IpfsB and published through pubsub to ipfsa
+              expect(res).to.exist()
+              expect(res).to.satisfy(checkAll([cidAdded, nodeBId.id]))
+
+              ipfsB(`name resolve ${nodeBId.id}`).then((res) => {
+                expect(res).to.exist()
+                expect(res).to.satisfy(checkAll([cidAdded]))
+
+                ipfsA(`name resolve ${nodeBId.id}`).then((res) => {
+                  expect(res).to.exist()
+                  expect(res).to.satisfy(checkAll([cidAdded])) // value propagated to node B
+
+                  done()
+                })
+              })
+            })
+          })
+        })
+      })
+    })
+  })
+
+  describe('disabled', () => {
+    let ipfsA
+    let nodes = []
+
+    // Spawn daemons
+    before(function (done) {
+      // CI takes longer to instantiate the daemon, so we need to increase the
+      // timeout for the before step
+      this.timeout(80 * 1000)
+
+      df.spawn({
+        exec: `./src/cli/bin.js`,
+        config: {},
+        initOptions: { bits: 512 }
+      }, (err, node) => {
+        expect(err).to.not.exist()
+        ipfsA = ipfsExec(node.repoPath)
+        nodes.push(node)
+        done()
+      })
+    })
+
+    after((done) => parallel(nodes.map((node) => (cb) => node.stop(cb)), done))
+
+    it('should get disabled state of pubsub', function (done) {
+      ipfsA('name pubsub state').then((res) => {
+        expect(res).to.exist()
+        expect(res).to.have.string('disabled')
+
+        done()
+      })
+    })
```

Comment:
Mocha supports promises. You should remove `done` and return the promise or use an async function. Mixing promises and callbacks is confusing.